eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -92,15 +95,37 @@ stages:
 
   # The Prerelease and Release stages are conditioned on whether we are building a pull request and the branch.
   - ${{if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal'))}}:
-    - template: archetype-java-release.yml
-      parameters:
-        DependsOn:
-          - Build
-        ServiceDirectory: ${{ parameters.ServiceDirectory }}
-        SDKType: ${{ parameters.SDKType }}
-        Artifacts: ${{ parameters.Artifacts }}
-        ${{ if eq(parameters.ServiceDirectory, 'template') }}:
-          TestPipeline: true
-        ArtifactName: packages
-        TargetDocRepoName: ${{ parameters.TargetDocRepoName }}
+    - ${{ if eq(parameters.EnableBatchRelease, 'true') }}:
+      - template: archetype-java-release-batch.yml
+        parameters:
+          DependsOn:
+            - Build
+          ServiceDirectory: ${{ parameters.ServiceDirectory }}
+          SDKType: ${{ parameters.SDKType }}
+          # By default, the Artifacts list will contain everything as we won't skip an artifact unless it,
+          # specifically, has releaseInBatch set to false. releaseInBatch should only be set libraries,
+          # not ready for release, in service directories that have batch release enabled. The if statement
+          # below, looking for releaseInBatch not equal false has to be set this way because if the library's
+          # metadata in the ci.yml doesn't have this variable then it defaults to being released. If the
+          # variable is set via checkbox, checked = true = release it, unchecked = false = don't release it.
+          Artifacts:
+          - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
Is there any reason this wouldn't work for both batch and non-batch cases? That way we can eliminate the duplication and just do the if condition around which of the templates to include but everything else will match?

Reply:
Trying this with the next commit.

Reply:
The answer to this is no. From everything I can see, the templates and parameters need to be in the same if/else block. Even tinkering around with the spacing didn't work. Having the parameters hoisted outside of the if/else for template selection made it such that none of the parameters were being passed.